src/Integration/ProfileConflicts/ConflictsManager.cs

```diff
@@ -138,7 +140,7 @@ private RuleSetInformation[] CheckSlnLevelConfigExistsAndReturnAllProjectRuleSet
             {
                 // Solution-level checks (done here because the expected solution-level config
                 // depends on the languages supported by the project that exist)
-                string baselineRuleSet = bindingConfiguration.BuildPathUnderConfigDirectory(ProjectToLanguageMapper.GetLanguageForProject(project).FileSuffixAndExtension);
+                string baselineRuleSet = bindingConfiguration.BuildPathUnderConfigDirectory(projectToLanguageMapper.GetAllBindingLanguagesForProject(project).FirstOrDefault().FileSuffixAndExtension);
```

Comment:
`GetAllBindingLanguagesForProject` can never return an empty list, as it uses the inner `GetLanguageForProject` that would return Unknown -- so it would be a list of 1 with "Unknown" in it.

In hindsight I think using `First` would be better than `FirstOrDefault`, LMK what you think and I'll update here and in the other places.

Reply:
In general, I think `First` would be better; if the list is empty there is something wrong in our code, so it would be better to fail so we can find the error and fix it.

Here, the projects have already been filtered, so we shouldn't get this far if the language is Unknown, should we?

Separately, both the old and new logic look suspect for C/C++ - the project supports two languages, but we are only ever checking one of them here. However, this doesn't seem to be a problem in practice - I've manually tested that the current code correctly reports when either C/C++ ruleset file is missing, and it detects both cases. I'm guess this is done later by the language-specific binding code.